test/index.js

```diff
@@ -1313,3 +1396,933 @@ test('contramap', function() {
 
   eq(Z.contramap(length, inc)('abc'), 4);
 });
+
+function testLaws(specs) {
+  Object.keys(specs).forEach(function(typeClassName) {
+    suite(typeClassName, function() {
+      var spec = specs[typeClassName];
+      Object.keys(spec.module).forEach(function(name) {
+        test(name.replace(/[A-Z]/g, function(c) { return ' ' + c.toLowerCase(); }),
+             spec.module[name].apply(spec.module, spec.laws[name]));
+      });
+    });
+  });
+}
+
+suite('laws', function() {
+
+  suite('Null', function() {
+    testLaws({
+      Setoid: {
+        module: laws.Setoid,
+        laws: {
+          reflexivity: [
+            jsc.constant(null)
```

Comment:
This works because `fantasy-laws` uses `sanctuary-type-classes` dispatchers, correct? That's a bit confusing, because essentially these tests are testing against the latest version of `sanctuary-type-classes` that is used in `fantasy-laws`. So if you would break something, these tests would not catch it until *after* you release a patch.

Reply:
That's a conundrum. Do you see a solution? Should fantasy-laws take `Z` as an argument? Should there be a peer dependency? Could we ensure correctness by following a certain procedure when making changes?

Reply:
One dirty, and partial, fix is to assign all the implementations to their respective fantasy land identifiers before running the tests, thus ensuring that the old `Z` dispatches to the code we want to test, eg:

```js
String.prototype['fantasy-land/concat'] = String$prototype$concat
```

This wouldn't work for `null` or `undefined`, though.

Reply:
I don't think a peer dependency will work, because I'm not sure whether npm will treat the module at `./` as an installed dependency.

Reply:
We could probably do something like `ln -sf . ./node_modules/fantasy-laws/node_modules/sanctuary-type-classes` before running the tests..

Reply:
> We could probably do something like `ln -sf . ./node_modules/fantasy-laws/node_modules/sanctuary-type-classes` before running the tests.

Good idea. This works. I have updated the pull request. :)